lib/src/wallet/transactions.rs

```diff
@@ -399,50 +477,53 @@ impl WalletTxns {
         height: BlockHeight,
         unconfirmed: bool,
         timestamp: u32,
-        nullifier: SaplingNullifier,
+        nullifier: WalletNullifier,
         value: u64,
         source_txid: TxId,
     ) {
         // Record this Tx as having spent some funds
-        {
-            let wtx = self.get_or_create_tx(
-                &txid,
-                BlockHeight::from(height),
-                unconfirmed,
-                timestamp as u64,
-            );
 
-            // Mark the height correctly, in case this was previously a mempool or unconfirmed tx.
-            wtx.block = height;
+        let wtx = self.get_or_create_tx(
```

Comment:
This fn should be `get_or_create_transaction_meta_data`, or something like that.

Reply:
`wtx` should be something like:  `spent_txn_metadata` or `spent_txn_data`